### PR TITLE
653: Add custom logger to redact parameters in prod-like environments

### LINKED ISSUE #653: custom TypeORM logger to redact parameters
Goal: Remove PII from logging. 

The alternative is to not do query logging in the application but do transaction logging on the database.

### PR DESCRIPTION
**Pull request by jlhogan**
closes #653

### CODE REVIEW

File: src/loggers/QueryLogger.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import { AdvancedConsoleLogger, QueryRunner, Logger } from 'typeorm';
```
Comment (bamader) [inline src/loggers/QueryLogger.ts:1]: Not an actual work comment, but the concept of an _advanced_ logger is hilarious to me. Like, it's not just `console.log()`. It's `console.*advanced*.log()` 😂

**Review by bamader — APPROVED** [review]
LGTM!

**Review by mmazanec22 — APPROVED** [review]

File: src/loggers/QueryLogger.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+  logSchemaBuild(message: string, queryRunner?: QueryRunner) {
+    super.logSchemaBuild(message, queryRunner);
+  }
+
+  logMigration(message: string, queryRunner?: QueryRunner) {
+    super.logMigration(message, queryRunner);
+  }
+
+  log(level: 'log' | 'info' | 'warn', message: any, queryRunner?: QueryRunner) {
```
Comment (JordanGuinn) [inline src/loggers/QueryLogger.ts:48]: Thoughts on adding at least `error` here, and potentially `debug` and `trace` as well?

**Review by JordanGuinn — APPROVED** [review]

File: src/loggers/QueryLogger.ts (hunk 2 above)
Comment (jlhogan) [reply src/loggers/QueryLogger.ts:48]: this is direct copy/paste from https://github.com/typeorm/typeorm/blob/master/src/logger/AdvancedConsoleLogger.ts

Comment (JordanGuinn) [reply src/loggers/QueryLogger.ts:48]: Oh weird - _why do they only support those three_?  Ah well 🤷🏽‍♂️

**Pull request merged**